dhis2-mobile-program-rules/src/main/java/org/dhis2/mobileProgramRules/RulesRepository.kt

```diff
@@ -41,9 +41,9 @@ class RulesRepository(private val d2: D2) {
             .let { orgUnit ->
                 orgUnit?.organisationUnitGroups()?.map {
                     if (it.code() != null) {
-                        supData[it.code()!!] = arrayListOf(orgUnit.uid())
+                        supData[it.code() ?: ""] = arrayListOf(orgUnit.uid())
```

Comment:
it's checking that `it.code() != null` one line above, maybe you can do
```
it.code()?.let { code ->
    supData[code] = arrayListOf(orgUnit.uid())
}
```
If you want to avoid the non null assertion